controlsfx/src/test/java/org/controlsfx/control/action/MemoryLeakUtils.java

```diff
@@ -0,0 +1,83 @@
+package org.controlsfx.control.action;
+
+import com.sun.management.HotSpotDiagnosticMXBean;
```

Comment:
I'm unsure about this import. Do we want to import these into ControlsFX? Will it cause issues when merged in JDK 11 branch?

The whole class is puzzling me. I did not know it was possible to test leak like that. Is it 100% sure? I thought the `System.gc();` output was unpredictable. I'm afraid this test adds 5 seconds to the build just to test for the listener. I'll let others review the changes.

Reply:
I'm using this technique in all of my projects.
Without it, it's basically guessing, which parts of the application are leaking.
It's by spec unsafe, but this is manageable and becomes a mere theoretical discussion.

In my experience, without these tests, JavaFX-code usually gets full with memory leaks, just like this one. Without these tests, memory leaks usually only get noticed in production, which is a not very systematic way to work on software.

The test should only add about 0.1 seconds on average. (there was a small bug, which I will fix tomorrow)



Actually, I've just started to make a library out of it, because I needed it for another. And I finally had to stop to copy-paste this logic from project to project.

Currently, I'm working on publishing this library on MavenCentral.

It probably would sense, when I would use my library in ControlsFX and add a lot of unit-tests to the library itself, to make sure this approach is working properly.

I can also provide a version of the PR without unit-tests.

Reply:
I'm sorry, I have lots of work. I'll take a deep look asap (maybe next week). I think it's better to the PR with the unit test, I just want to be sure of what we have. But if we encounter some issue in JFX13 branch or 9.0.0 branch, then we will maybe remove it for this specific branch